### DISCUSSION TITLE
Nuxt 3 Release Candidate

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#18383] [Announcements]
[👉](https://emojipedia.org/backhand-index-pointing-right/) **[Check out the Nuxt 3 roadmap !](https://v3.nuxtjs.org/community/roadmap)**

## Update: [RC.1 is out!]https://nuxtjs.org/announcements/nuxt3-rc/

You can join [this discussion](https://github.com/nuxt/framework/discussions/4479).

## Intro

Since the [public announcement of Nuxt 3](https://nuxtjs.org/announcements/nuxt3-beta), thanks to the help from all beta testers and contributors, we were able to collect more feedback, improve stability and prepare the ecosystem for Nuxt3 💚 

The main goal of initiating the release candidate stage is to **increase the adoption rate of Nuxt 3**, and **change stability** (not fixing all remaining issues but introducing fewer regressions!).

~~The expected release date for the first release candidate is **March 2022**. Steps for reaching `nuxt@3.0.0-rc1` are explained and trackable here.~~

## 📅 **Update 2:** Another release date

👍 The _Good_ news is, that the [unjs/nitro](https://github.com/unjs/nitro) migration finally landed via giant #3956 PR! It gives nuxt3 a solid foundation for server support in the future.

⌛ The other news is, that we need to wait a little longer for `nuxt@3.0.0-rc.1`. But it isn't that far and we do hope to announce the first release candidate during the upcoming week! **Update: Set to April 20, 2022**

In the meantime, we will keep polishing the 📚 documentation, fight the 🐛 bugs also publish an initial release as `nuxt@3.0.0-alpha` in order to verify package name changing from `nuxt3` is fine across the ecosystem and not conflicting with `nuxt@2.x`. 

At this point, we truly value your **patience**, feedback by **testing nuxt 3 edge releases**, help to **identify any major issue** and of course **contribute and improve documentation**.

## 📅 **Update 1:** New release date

We are doing our best to deliver RC.1 as soon as possible. Due to recent major changes in the codebase and docs, postponing release to Thursday ~~**April 7, 2022**~~ in order to verify improvements against `nuxt3` edge releases and ensure there are no last-minute regressions.

- Nitropack (#3956, #3161)
- `vite-node` (#3598, #2795)
- Migration Guide (#3819) ([feedback discussion](https://github.com/nuxt/framework/discussions/3989))

 
## Framework Tests

With a solid, stable test suite, we can push changes with more confidence and fewer regressions when upgrading the Nuxt version.

Checklist:
- Stable tooling to add more tests (`@nuxt/test-utils`) (https://github.com/nuxt/framework/issues/3198)
- Have at least one test for **each documented feature**
- DX tests for development mode (`nuxi dev`)

## Documentation

Before reaching RC, we should add **stability labels** to each section, identify **undocumented parts**, and improve **migration guides**.

Current documentation uses outdated tooling based on Nuxt 2. We will soon upgrade it to a Nuxt 3 theme and Content v2.

Checklist:
- Update [v2 to v3 migration guide](https://v3.nuxtjs.org/getting-started/migration) with more details and feature compartion (full steps might not be possible with RC.1)
- Identify undocumented parts
- Add stability labels for each section
- Get started guide
- Releases page
- Migrate to Nuxt 3 and Content v2

## Release Workflow

We have been pushing some breaking changes so far without clear notices to iterate faster. A commutative changelog of possible breaking changes introduced in Nuxt 3 is listed here [nuxt/framework#2883](https://github.com/nuxt/framework/discussions/2883).

Starting with release candidates, the process of releasing will be the same as Nuxt 2, with carefully crafted notes and additional checks. Automated edge release channel will still be available for beta testing changes before each RC release.

Checklist:
- Prepare changelog and release generation scripts
- Refactor to release as `nuxt@3` and `nuxt-edge@3` npm packages
- Set release schedule policy for RC

## Finalize composables

We had been experimenting with several new ideas with composables. [data-fetching composables](https://v3.nuxtjs.org/docs/usage/data-fetching) can possibly have major changes which we should investigate before releasing RC.1 to either introduce the breaking change or add proper warnings to the documentation.

## Ecosystem

A healthy ecosystem is not less important than a stable Framework. Therefore, we shall ensure vital modules are ready to be used with RC.1 to make the adoption and migration process as accessible as possible.

Checklist:
- Image module (https://github.com/nuxt/image/issues/500)
- Telemetry module (https://github.com/nuxt/framework/issues/303)
- Review current [Nuxt 3 compatible modules](https://modules.nuxtjs.org/?version=3.x) following best practices and ready for RC.1

## Package externalization

To keep the new [nuxt/framework](https://github.com/nuxt/framework) beautiful and maintainable for a long-term period, we need to make **core as tiny as possible**. We had been doing this progressively since Nuxt 2 for popular packages like `consola`, `hookable`, `std-env`, `ufo`, `jiti` and `webpackbar`, which not only helped to stabilize Nuxt 2 with fewer maintenance efforts but also benefit the rest of the JS ecosystem. Moreover, we had accelerated this with [unjs](https://github.com/unjs/) project during Nuxt 3 development.

We need to finish externalization before reaching the release candidate since this is a time-consuming and blocker process and often causes regressions.

Checklist:
- Nitropack (https://github.com/nuxt/framework/issues/3161
- Autoimports with `unimport` (https://github.com/nuxt/framework/pull/3386)
- CLI builder

**Comment by dilikpulatov** [comment]
Finally. Thanks guys for your work)

**Comment by Baroshem** [comment]
For the ecosystem I would also add this PWA module as it is used widely in many projects nowadays.
https://github.com/nuxt-community/pwa-module/issues/492

Apart from that, I completely agree with the whole checklist. Can't wait to see the final result. A lot of great work was done here.

Congratulations guys!

**Reply by pi0** [reply]
Thanks, @Baroshem! Indeed PWA is included in the roadmap: https://v3.nuxtjs.org/community/roadmap#%EF%B8%8F-roadmap We are thinking of merging it to the Nuxt 3 core as it gives much better possibilities with server worker integrations and certainly more stability.

**Comment by ivaneliasoo** [comment]
Great work thank you! we're looking forward!

It would be great if possible, to add nuxtjs/auth module to the checklist

**Reply by pi0** [reply]
We are planning to initiate a fresh rewrite of Auth module based on session support and composables from Nuxt 3. It is included in roadmap: https://v3.nuxtjs.org/community/roadmap#%EF%B8%8F-roadmap

**Comment by ohuu** [comment]
This is great news, many thanks to all the contributors! Can I ask what the status of static site generation is for the latest version and RC1? Can we generate static sites with version 3 yet?

**Reply by pi0** [reply]
It is included in roadmap for this month too! https://v3.nuxtjs.org/community/roadmap#%EF%B8%8F-roadmap

**Comment by webfansplz** [comment]
Great work thank you ！

By the way , I personally think it would be better if we had nuxt/nuxt#17750 😃

**Reply by Aareksio** [reply]
I find this **very** important. Error handling is inconsistent, to the point we can not _properly_ display custom 404 page with HTTP `404 Not Found`. Setting response status code (common examples being 401, 404, 429, 503), is either not supported, or requires knowledge of non-standard dependencies (eg. h3 instead of express). It requires at least documentation.

**Reply by pi0** [reply]
We are working on it at the same time. See https://github.com/nuxt/framework/pull/3464

**Comment by toniengelhardt** [comment]
Tracking the community module ecosystem here: https://repo-tracker.com/c/nuxt3 (made with Nuxt 3)

If you can specify the list with all "vital" modules, I can make another list for them 🙏🏽

**Reply by pi0** [reply]
This is suer cool with repo-tracker 🔥 

They are listed in the roadmap (https://github.com/nuxt/framework/pull/3464) but most of their rework is either in another branch or a private repository so probably not giving best results at the moment to track their activity.

**Reply by toniengelhardt** [reply]
Ok, then I'll keep the more extended list with the most popular modules, thanks 🙏🏽

**Comment by warrior-dev** [comment]
Thanks, guys for your effort and passion ✨

**Comment by userquin** [comment]
Hey nuxt team, great work, just a simple question: where will be `pnpm` support available again? It seems latest versions stop working, check the vitesse nuxt 3 repo issue here https://github.com/antfu/vitesse-nuxt3/issues/21

**Comment by levimichael** [comment]
Thank you so much for the update and transparency! This is what we needed!

**Comment by olliepotter** [comment]
Can’t wait to use Nuxt 3! Only reason I didn’t upgrade was because it wasn’t ready for production. Great work guys! 👏🏼How can we support the framework? Do you guys take donations? If so where abouts?

**Reply by harlan-zw** [reply]
You can make a contribution here: https://github.com/sponsors/nuxt or https://opencollective.com/nuxtjs

**Comment by jerryjappinen** [comment]
Great to see some progress on this front.

Was just checking if the latest version had fixed #3589 but found a new build error instead (`Cannot read property 'includes' of null` in `node_modules/nuxt3/dist/chunks/index.mjs:313:29`, broken in `3.0.0-27449789.5d58ef4` I think).

Tried to set up a new minimal repo with a reproduction so I could file an issue. Only to find out that a new, blank app straight out of `nuxi init nuxt3-app` will actually not build on `latest` (`[vite:build-import-analysis] Parse error @:1:47` on `v3.0.0-27451165.114cbe3`).

Are there any plans for different branches/tags for more stable builds during beta/RC phase? I think it's valuable to give early adopters a little bit more of a safe environment, keeping in mind that a breaking build prevents the developer from testing and providing feedback on other areas of the tech and DX.

**Reply by danielroe** [reply]
I'd suggest filing an issue with a reproduction, and I'd be happy to look into it, but I can't replicate what you describe myself.

I believe the section in the post above marked 'Release Workflow' answers your questions. (tl;dr: Yes.)

**Reply by jerryjappinen** [reply]
Yeah, I saw that and that would be welcome. It's paired with a refactor and release in `nuxt` though so I was wondering if that's still a while away?

**Comment by LoiLock** [comment]
Any updates?

**Reply by avxkim** [reply]
Looks like they won't be able to release first RC version in march

**Reply by pi0** [reply]
Don't worry. We are working actively and hope to make this happen! Items in the checklist of the discussion above are only informative of what is expected with RC.

**Reply by PDieE** [reply]
> Don't worry. We are working actively and hope to make this happen! Items in the checklist of the discussion above are only informative of what is expected with RC.

Any date updates？

**Reply by LoiLock** [reply]
He gave us an update yesterday?

**Reply by a1xon** [reply]
@pi0 Please don't rush it. Stability > Punctuality.

**Reply by kissu** [reply]
Exactly! Also, trying to be sane at the end of the RC could be a nice thing. ❤️

**Comment by danpastori** [comment]
I noticed there was a migration guide mentioned (https://v3.nuxtjs.org/getting-started/migration/). I've been in the process of updating our app from Nuxt 2 to Nuxt 3 entirely. I know Nuxt 3 is still in beta and under active development, but would like to contribute in some way. 

Is there a format where I could add some migration documentation? Or should I wait until Nuxt 3 is fully released?

**Reply by danielroe** [reply]
@danpastori I'm working on the migration guide: https://github.com/nuxt/framework/issues/3769. Perhaps connect with me on Discord and we can work together?

**Reply by danpastori** [reply]
@danielroe Perfect, just hopped on. Shoot me a message and I can help anywhere I can

**Comment by tanfabbi** [comment]
Did you guys find the official nuxt release too slow?

**Reply by kissu** [reply]
Not always simple to keep the pace with a deadline. Happens to all of us! 😺

**Reply by tanfabbi** [reply]
I think, by the time the official version is released, Vue has also updated the new version.

**Reply by kissu** [reply]
I'm not *sure* it will not. 😉
Also, incremental updates are easier once everything is migrated to Vue3 so it should not be a problem. 👍🏻

**Reply by benoitdemaegdt** [reply]
@tanfabbi  we are all looking for the official release. But please show some respect to the amazing team behind Nuxt. They are working very hard these days. Have a look at the commit history to see this 👀 

If it really takes too much time for you, here are some tips to make the process faster :
- Contribute to Nuxt3 (Open a PR, update the documentation, help people in github discussions ...) 💪 
- Sponsor the Nuxt Team so that they might be able to hire another software engineer 💰 . (Here is the [github sponsor link](https://github.com/sponsors/nuxt))
- Start building your project with the current version. There might be small changes, but probably no major change in coming days.

Happy coding 🎉 🙌 
And thank you to the Nuxt team 💚

**Reply by ohuu** [reply]
@tanfabbi it's an open source project, you get to use it for __free__ without having to lift a finger. If the release is too slow why don't you roll up your sleeves and start contributing? Honestly the level of disrespect that some people have for open source projects and the people that contribute to them never ceases to amaze (and annoy) me.

Thank you to the nuxt team and all open source contributors, you guys rock.

**Reply by kissu** [reply]
Yeah, OSS work is often taken for granted unfortunately. 😅

**Reply by instanceasia** [reply]
> @tanfabbi it's an open source project, you get to use it for **free** without having to lift a finger. If the release is too slow why don't you roll up your sleeves and start contributing? Honestly the level of disrespect that some people have for open source projects and the people that contribute to them never ceases to amaze (and annoy) me.
> 
> Thank you to the nuxt team and all open source contributors, you guys rock.

What are you thinking? Criticize then say don't contribute? Don't you think it's slow? I'm giving you an idea to speed things up, but if you feel hurt you can ignore it.

**Comment by PDieE** [comment]
Can it be released this month?

**Comment by dommi10** [comment]
Only 3 days before april.
Still waiting for Release Candidate (RC)

**Reply by LoiLock** [reply]
Some of you guys are incredibly impatient...

**Reply by dommi10** [reply]
Yes
I stopped to create new projects, I’m waiting for RC to start again

**Reply by kmuip** [reply]
That's a pretty weird approach tbh

**Reply by itsmnthn** [reply]
Waiting ✋️ for RC too

**Reply by kissu** [reply]
It's coming pretty soon! 😺 
Here's a cute GIF to ease the wait!!

![](https://c.tenor.com/U7h-gyy--akAAAAC/kiss.gif)

**Reply by ohuu** [reply]
@dommi10 instead of complaining about it why don't you contribute and support the project?

**Comment by maxdzin** [comment]
Hi guys!
I have been trying the Nuxt v3 for some time and now working on moving some projects codebase into it and I want to say that you making amazing work for v3. I'm looking towards RC and hope we can try it soon. And thank you so much!

**Reply by pi0** [reply]
Thanks! and happy to hear you are already moving your codebases to Nuxt3 💚

**Comment by bros4president** [comment]
Thanks guys for the great work! We're about to start coding a new project in vue3 + nuxt3, hoping RC will be ready when we're finished.

One question we have not been able to answer for ourselves:

Why is [Vite being labeled as experimental](https://v3.nuxtjs.org/getting-started/introduction#comparison)? Does this mean:
a) Not working
b) Trying to make it work, but may not be supported in the future
c) Partially working
d) Fully Working, but usage may change drastically
e) Other

We're really looking forward to use Vite in this project and it's the only reason we have not started implementing nuxt yet.

Thanks!

**Reply by Aareksio** [reply]
Vite itself is fully functional. Hot reloading is giving way more trouble than webpack's, but it compensates with the speed. Your mileage may wary depending on complexity and quality of your code and 3rd party plugins - not every behavior from webpack is easily replicable with Vite. Nuxt 3 is not plug-and-play with plugins like Nuxt 2 as of yet. I wouldn't expect it to be during the RC1 either - prepare to get your hands a bit dirty and you'll be fine.

**Reply by anburocky3** [reply]
@Aareksio Yea, regarding HRM, i was working with tailwind css and while changing my html code, sometimes, it doesn't hot-reload. Is this the issue you are talking about?

**Reply by dosstx** [reply]
I have seen this too. Now I have a bad habit of always manually refreshing my browser in case....but this is not specific to Nuxt3 because I haven't used Nuxt3 yet and I am just playing around with Vue 3 and tailwind.

**Reply by anburocky3** [reply]
Oh, maybe with `vite`? But anyhow, this process is irritating. 😔

**Comment by romainl** [comment]
Hi, with RC1 just around the corner, would it be possible to extend [the v3 roadmap](https://v3.nuxtjs.org/community/roadmap/) to include official plans for v2?

As of now it is not clear whether v2 will still receive *some* kind of support (security fixes, critical bugs, etc.) after the official release of v3 and, if it does, for how long.

In other words, when and how do you plan to EOL v2?

**Reply by pi0** [reply]
We keep adding hotfixes and security updates to Nuxt 2 and backporting a subset of useful features using `@nuxt/bridge`.

Even by keeping our best to support Nuxt 2, old dependencies like `webpack@4` will eventually break Nuxt 2 and there is not much we can do about dependencies. So we urge users to consider migrating to Nuxt 3 stable release. 

BTW thanks for the feedback! We definitely need to make it more clear in the roadmap.

**Reply by romainl** [reply]
Thank you.

We are going to set up a new Nuxt project next week. We were initially planning to do it with v2, which is stable enough and with which we have a few years of experience at this point. The little conundrum we are in is that v3's planned release is going to happen in the middle of the project so we are wondering whether it makes more sense to:

- start with v2 and switch to v3 after release (possibly with the bridge right from the start to ease the transition),
- or to start with v3 right away, which—as of today—seems to be a little rough around the edges and has limited module support.

Thanks.

**Reply by pi0** [reply]
At this stage, I would invest in Nuxt 3 to start a new project. You have a faster dev experience and you can always directly add integrations to your nuxt project that are missing a Nuxt 3-ready module.
 
Nuxt bridge is an in-between solution. By using Nuxt 2, you are opting into an ecosystem of old and unmaintained packages. I would only do this for an existing project or if my new project can heavily reuse existing developed codes and modules.

**Reply by dosstx** [reply]
I had the same question and your response was informative. Thanks, @pi0

**Reply by toniengelhardt** [reply]
@pi0 I think the reason a lot of projects are stuck on v2 is that they rely on the `auth` and `pwa` modules, which are not so easy to replace and which are not yet compatible with either Bridge or v3.

**Reply by pi0** [reply]
PWA module will be part of core with built-in meta and a unified service worker system. And auth to be rewritten using a new `useSession` composable.

Will be back to work on both and hopefully land a basic version in April after RC. 

In the meantime, you can add both service worker/meta tags and implement auth using server middleware and nuxt 3 composables. Modules finally only make the job easier :)

**Reply by romainl** [reply]
Thank you very much, @pi0.

**Comment by pi0** [comment]
After several improvements and fixes here and there, we finally decided to postpone the release by one week to Thursday, **April 7, 2022** for a stable and better-tested release.

Please check out the updated description & Stay tuned 🔥

**Reply by amirnbcode** [reply]
Good News. 
I've already started a Nuxt3 project. looking forward to the new release. Thanks for this beautiful artwork.

**Comment by zhangkai0811** [comment]
莫道谗言如浪深，莫言迁客似沙沉。
千淘万漉虽辛苦，吹尽狂沙始到金。

**Reply by kissu** [reply]
Not sure to fully understand, but I guess that it's kind so thanks! 🙏🏼

**Reply by Benbinbin** [reply]
it said many people say a lot (complaints), but your guys still keep working, and finally reach the RC.1 (金 mean gold, here means something valueable)

**Comment by lvreninfo** [comment]
Looking forward to see [ant-design-vue](https://github.com/vueComponent/ant-design-vue)、[element-plus](https://github.com/element-plus/element-plus) 、[vant UI](https://github.com/youzan/vant) working with Nuxt 3 in the near future
Any news about this ??

**Reply by miscab** [reply]
Just one day left to see the release candidate, I hold on all the projects on design phase right until this moment.

**Comment by miscab** [comment]
I bet the team will hold up to 23:59:59, with 1 hour bias. 😄

**Reply by Celibioux** [reply]
we all hope that the RC will be today ...but really guys @pi0 @Atinux @danielroe @antfu and all the team are doing the greatest work ever ^^ so no worries if we will have all this goodness in Nuxt (even if i check every day for the good news heheh )

**Comment by sea-lanya** [comment]
I'm waiting for ...

**Reply by zhangkai0811** [reply]
Me too

**Reply by LeoCpii** [reply]
Me too 👀

**Reply by yashraut362** [reply]
Any update on the stable release?

**Reply by chantouchsek** [reply]
> Any update on the stable release?

Not for stable for today.

**Reply by kissu** [reply]
It's coming guys, don't spam. 😄 
Subscribe to the twitter accounts of [Nuxt](https://twitter.com/nuxt_js) and [Atinux](https://twitter.com/Atinux) and you should be good to go. 👌🏼

**Comment by chantouchsek** [comment]
I've been waiting..... from 🇰🇭

**Reply by Foddy** [reply]
🇩🇪🕡…

**Comment by olliepotter** [comment]
Just want to remind the people who are working on this that everyone that are seemingly impatient are just so excited because you've built something so awesome and we all want it now! 🚀

... so hurry up 😏

**Comment by kissu** [comment]
Hey everybody! 👋🏼 
As everybody here knows, the RC was planned for the 7th. As you can see, nothing was announced yet.
Keep in mind that the core team was still fixing things today (April 8th) at 2am aka doing their best rather than sleeping. 😴 

So yeah, a friendly reminder to say that people at NuxtLabs are human beings (some have a family too), so please be respectful of their efforts in the next hours! 💛💛
It will mean a lot to them and will be a good mental support to see that you can hold on just a bit longer! 🙏🏼

**Reply by Celibioux** [reply]
@kissu the best comment i've seen really ! 
we should be more reminded that eveyone in the NuxtLabs are doing all the best to deliver something great to us to help us and of course we all wait because we love Nuxt and all the good stuffs with it ! but we should respect all the folks who works super hard 
Thank you for Morocco ^^ to all those help making Nuxt the Best i've ever used

**Reply by kissu** [reply]
Disclaimer: this is not my actual face but I'm sending the same vibes.

![](https://c.tenor.com/7xFbk82J5W0AAAAd/thank-you-for-your-support-vanderpump-rules.gif)

**Reply by jskolovos** [reply]
I agree. I know the core team is doing their best efforts. I'm seen activity in the repository. Just happens they are still working on it.
We are eager to see the release candidate, but we can wait if the team feels it is not ready yet.

**Reply by Celibioux** [reply]
that's true @jskolovos well said bro as much as we love Nuxt for every cool stuff it gives us and make it easer to work in any project i think that is not enough to thanks all the genuis behind it all support to @pi0 @antfu @Atinux @danielroe and all the folks (sorry if i've forgot anyone) and i think a good working late software million times better than a buggy earlier one

**Reply by tobiasdiez** [reply]
While I agree with the general sentiment and think everyone here is doing an awesome job building nuxt, one shouldn't forget that its their normal day-to-day job. They get paid for it and nuxt is not a hobby project that the work on in their freetime. Similarly, nuxtlab is a "proper" company. Thus I think it is also justified to expect a certain standard of communication and professionalism. 

A good comparison would be maybe vscode. Microsoft is developing it as open-source software, but at the same time adheres to regular release plans and communicates well with proper changelogs and public roadmaps. From what I've seen the nuxt team is heading in a similar direction, so let's keep the momentum going 🚀.

**Reply by atinux** [reply]
The framework team has been working for free for a long time on Nuxt before we (NuxtLabs) got the opportunity to sponsor them to give the opportunity to dedicate all the time on the framework development. This does not mean they should work differently, maintaining and developing a framework is mentally exhausting and time consuming.

I appreciate the comparison with Microsoft 😊  but we are far from having the resources they have (marketing, communication, testers), also I don't know how they were doing when VS Code was in alpha/beta.

Release notes and communication are easier once you hit a stable release, right now Nuxt 3 is moving really fast to reach the release candidate. I am currently writing the article for the RC at the moment to be as transparent as possible toward the framework status.

Thank you very much for your patience, it's stressful and exciting at the same time 😄 

Thank you @kissu for your message ❤️

**Comment by dosstx** [comment]
Soon as I saw my github notification with `Atinux` in the message, I clicked the link thinking that it was an announcement that Nuxt3RC is released! But, no problem...take your time. I wanted to say that I totally understand the "mentally exhausting" experience. I'm going through that phase right now with a project and it's at the point now where I am starting to suffer  physically (lack of exercise, etc). Make sure the Nuxt team takes mental breaks and get some daily physical activity in. It may not feel like it, but taking breaks actually speeds up the development time of projects more than we think. Keep up the good work!

**Comment by pi0** [comment]
## 📢 Second update

> 👍 The _Good_ news is, that the [unjs/nitro](https://github.com/unjs/nitro) migration finally landed via giant #3956 PR! It gives nuxt3 a solid foundation for server support in the future.
>
> ⌛ The other news is, that we need to wait a little longer for `nuxt@3.0.0-rc.1`. But it isn't that far and we do hope to announce the first release candidate during the upcoming week! 
>
> In the meantime, we will keep polishing the 📚 documentation, fight the 🐛 bugs also publish an initial release as `nuxt@3.0.0-alpha` in order to verify package name changing from `nuxt3` is fine across the ecosystem and not conflicting with `nuxt@2.x`. 
>
> At this point, we truly value your **patience**, feedback by **testing nuxt 3 edge releases**, help to **identify any major issue** and of course **contribute and improve documentation**.


Cheers, and wish you a lovely weekend on the nuxt team's behalf!

**Reply by zweiter-hub** [reply]
I really like nuxt, but nuxt 3 seems like a revolution to me, because the things I hate in nuxt2 were fixed in nuxt3 (or made better). I really want you to announce your stable version. How can I contribute to speed up the process?

**Reply by dubem-design** [reply]
Am not sleeping too till we go live !!!,

Weldone guys.

**Reply by kissu** [reply]
@Segundo-hub if you want to help: https://github.com/nuxt/framework/discussions/3447#discussioncomment-2506308

**Reply by matt88120** [reply]
Is the RC will delivered static site generation ?

**Reply by kissu** [reply]
@matt88120 this is part of the PR (above) that got merged yeah. 👍🏻

**Reply by matt88120** [reply]
@kissu Any docs about it ? I only see SSR boolean, but it seem it have no effect

**Reply by kissu** [reply]
@matt88120 it looks like it's not out yet, probably being written as of right now. Or will be presented shortly.
You can get some details here tho: https://v3.nuxtjs.org/guide/concepts/server-engine

**Reply by lukas-a-brand** [reply]
Thanks for all the hard work there, folks! All the blood, sweat, and tears will be worth it! (:

**Comment by StevenJPx2** [comment]
Hey, hope it releases soon. Just love the batteries-included but minimal approach that Nuxt 3 takes which Vitesse cannot do! 

I think the most important thing is to take care of aged P4 and P5 before ever releasing an RC. Hope you guys are working on that. I am trying to figure out and fix those bugs myself, but I can't really navigate through the code that well! I'll try my best though, so the RC can come out as soon as possible!

**Comment by flotothemoon** [comment]
Thanks for the great work on Nuxt. I would like to use Docus/Nuxt Content v2 for a new Nuxt3-based website, but couldn't find any information on the planned release for the content v2 module (the Docus Twitter and website haven't been updated in a while). 

Do you have a timeline for when Docus/Content v2 will be made available to the public? If it's longer away, what do you suggest a Nuxt3-based website do in the meantime for its content?

**Reply by doostinharrell** [reply]
Content v2 is on the roadmap for Q2 of 2022. I’m holding out on developing any Nuxt 3 projects until that is ready as well. https://v3.nuxtjs.org/community/roadmap

**Reply by kissu** [reply]
Docus is currently being worked on. 👌🏻
As shown in the [roadmap](https://v3.nuxtjs.org/community/roadmap#%EF%B8%8F-roadmap), this is planned for Q2 still.

Still, at the end you will probably write some markdown I presume, so the change will not be that big of a leap.

**Reply by flotothemoon** [reply]
Got it, thanks for the answers @doostinharrell and @kissu. 

> Still, at the end you will probably write some markdown I presume, so the change will not be that big of a leap.

How would I render Markdown properly in Nuxt3 without the content framework? Or are you suggesting just writing the Markdown first without Nuxt3? 

Also, if there's any way to contribute to Content v2, I would be happy to help out (I've already signed up for the Docus beta).

**Reply by atinux** [reply]
We are doing the final tests for Nuxt Content v2 to work with latest version of Nuxt 3 and Nitro, our deadline is end of April to open source it 😊

**Reply by flotothemoon** [reply]
Awesome @Atinux, will be happy to contribute where I can :)

**Comment by chenshunliang** [comment]
When will the Nuxt 3 Release Candidate be released ?

**Reply by kissu** [reply]
Hi, please check this one: https://github.com/nuxt/framework/discussions/3447#discussioncomment-2530480

And this one: https://github.com/nuxt/framework/discussions/3447#discussioncomment-2530568

**Reply by stepaniukm** [reply]
RC date updated to the April 20th

**Reply by kissu** [reply]
I confirm. 👍🏻

**Reply by HendrikJan** [reply]
Thank you for the update 👍

**Comment by Foddy** [comment]
Developers around the world today:

![image](https://user-images.githubusercontent.com/5302050/164166669-4e569ed1-15cd-4932-b291-7f9ea3c6f3fb.gif)

**Comment by sikeat7** [comment]
Waiting for RC release but I think not only me!
Appreciated to NuxtTeam 🤪

![](https://c.tenor.com/K_zx7rHseTAAAAAd/excited-hockey.gif)

**Reply by avxkim** [reply]
RC release is postponed till 27...

**Reply by reinoldus** [reply]
What's your source?

**Reply by avxkim** [reply]
@reinoldus you'll see, they will announce soon.

**Reply by pi0** [reply]
Common we won't postpone again 😄 But let's plan another release for next week ;)

**Comment by amirmms** [comment]
![common ](https://media.makeameme.org/created/just-waiting-for-dfbe76cfed.jpg)

**Comment by mctweb** [comment]
It's live!
(Just about... )

**Reply by Foddy** [reply]
And I lost my bet.

**Reply by avxkim** [reply]
I bet they will postpone it again!

**Comment by alvarosabu** [comment]
Amazing work @pi0, I really admire how the team manages to add so much value to the framework.

With the release candidate will be possible to generate a static site generation (SSG) on Netlify for example?. Similar to `nuxt generate` at the previous version https://github.com/nuxt/framework/issues/3161  https://nuxtjs.org/docs/concepts/static-site-generation/

I tried 2 weeks ago and the preset for Netlify was generating an SSR instead (even with the properties on nuxt.config).

Thanks in advance

**Reply by anburocky3** [reply]
+1

**Comment by SamRahnama** [comment]
![mr-bean](https://user-images.githubusercontent.com/18219117/164223320-2e51563c-3991-43f0-8718-173306b8c8e0.gif)

**Reply by adarshmadrecha** [reply]
![WaitKittenGIF (2)](https://user-images.githubusercontent.com/11911938/164223657-6eaa428e-06d0-40c2-a8cc-676c1058bf65.gif)

**Reply by avxkim** [reply]
hope you got lot of patience xD

**Reply by wujingquan** [reply]
https://www.npmjs.com/package/nuxt/v/3.0.0-rc.0

**Comment by maxdzin** [comment]
So, magic is coming ))
Big love and thanks to all the team for working hard!

https://www.npmjs.com/package/nuxt/v/3.0.0-rc.0
